src/ui/views/Send/SendToCadence/index.tsx

```diff
@@ -42,78 +46,56 @@ const EMPTY_COIN: CoinItem = {
   icon: '',
 };
 
-const SendAmount = () => {
+const SendToCadence = () => {
   const history = useHistory();
   const location = useLocation<ContactState>();
   const usewallet = useWallet();
-  const { childAccounts, currentWallet } = useProfileStore();
-  const [userWallet, setWallet] = useState<any>(null);
-  const [currentCoin, setCurrentCoin] = useState<string>('flow');
-  const [coinList, setCoinList] = useState<CoinItem[]>([]);
+  const { childAccounts, currentWallet, userInfo } = useProfileStore();
+  const { coins: coinList } = useCoinStore();
```

Comment:
I've had issues previously in using arrays as dependencies. I could be doing it wrong... but I'd do this:

```suggestion
  const coinStore = useCoinStore();
  // ... 
  const coinInfo = coinStore.coins.find((coin) => coin.unit.toLowerCase() === token.toLowerCase());

```